src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8131,6 +8131,28 @@ public static string GetTarget(PSObject instance)
             return null;
         }
 
+        /// <summary>
+        /// Gets the target for a given file or directory, resolving symbolic links.
+        /// </summary>
+        /// <param name="instance">The FileInfo or DirectoryInfo type.</param>
+        /// <returns>The file path the instance points to.</returns>
+        public static FileSystemInfo ResolvedTarget(PSObject instance)
+        {
+            if (instance.BaseObject is FileSystemInfo fileSysInfo)
+            {
+                FileSystemInfo symbTarget = fileSysInfo.ResolveLinkTarget(true);
```

Comment:
What is a negative effects from the throw? Should we really mask the exceptions?

Reply:
When calling the method on a file on a USB drive or micro sd drive, it will fail because the underlying filesystem doesn't support reparsing point. However, the behavior in that case should be returning the full path of the file.

Actually, this scenario should be handled, meaning when exception happens, it should decide when to return the `FullName` of the current `FileSystemInfo`, and when to return `null`.

Reply:
This would weird behavior in formatting output - if user see Target and LinkType  properties are "empty" but ResolvedTarget is FullName.
And if I remember correctly .Net ResolveLinkTarget can return partially resolved path. https://github.com/dotnet/runtime/blob/12a8819eee9865eb38bca6c05fdece1053102854/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs#L621-L627

Reply:
The behavior I describe is what was asked from the issue:

> If the item at hand isn't a symlink / reparse point, simply report its own full path - this allows unified treatment of symlinks and non-symlinks, as a simple way to get an item's full path.

Reply:
I'm about to push a commit that wraps a try-catch around the `ResolveLinkTarget` call and returning `null`, however I'm not sure in what situation `ResolveLinkTarget` would throw an exception and want to return `FullName`. It cleanly returns `null` when encountering something that isn't a symlink, and flash drives (by flashdrives, I'm assuming you mean the usage of FAT32/exFAT here) don't seem to support symlinks in any capacity. Am I missing something here? Thanks!

Reply:
I think we want to return `FullName` only if the `FileSystemInfo` is just a regular file/directory instead of a link.

Looking more into it, I found on portable drive with exFAT file system, `ResolveLinkTarget(false)` throws while `ResolveLinkTarget(true)` doesn't throw ([see code here](https://github.com/dotnet/runtime/blob/09c1a1f7b0c477890b04912d8dd4f742f80faffc/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs#L417-L425)). So, maybe we don't really need the `try / catch (IOException)` here.

So, maybe we can do the following:
1. if `ResolveLinkTarget(true)` returns `non-null` value, just return that value. (I see you check on `symbTarget.Exists` here, but could it even be possible that the method returns a non-null value, but it doesn't exist?)
2. if `ResolveLinkTarget(true)` returns `null` value, assuming that means it's not a link, and thus return the `FullName` or the current `FileSystemInfo` depending on what we want for return type.
3. if `ResolveLinkTarget(true)` throws, then we assume that means it could be link, but something is wrong when resolving it. We just let the exception propagate in this case.

Reply:
Based on the docstring for `ResolveLinkTarget`, it looks like we do need to check if it exists:

> Returns: A System.IO.FileSystemInfo instance if the link exists, independently if the target exists or not

Besides that, the commit I mentioned earlier and the ones I just pushed should match the requested behavior!